src/io/encoding.cr

```diff
@@ -184,10 +184,10 @@ class IO
 
       # We need to read from the out_slice into a String until we find that byte,
       # or until we consumed limit bytes
-      String.build do |str|
+      String.build do |io1|
```

Comment:
use `str` is really not correct, because it never be a String, it is a IO like object, that really not same things, it will mislead the code reader.

i rename it into `builder`, it is better?

> The diff is pretty big, which will lead to merge conflicts. I don't know if that's worth it.

Only rename but never change logic, high probability will not lead to any conflict.

> Also, you sometimes renamed the block parameters to "buf" in your PR (instead of "io" in the majority of times). Was that by accident or is there a specific reason?

Because original use `buffer` as block parameter in specific context, that at least better than `str`, i just rename it into a shorter name `buf`.

Reply:
Hi, i rename |buffer| or |buf| in String.build into `|io|` too, feels more uniform.